src/Core/Platform/Infrastructure/Repository/LegacyReadVersionRepository.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Platform\Infrastructure\Repository;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Core\Platform\Application\Repository\ReadVersionRepositoryInterface;
+use Symfony\Component\Finder\Finder;
+
+class LegacyReadVersionRepository extends AbstractRepositoryDRB implements ReadVersionRepositoryInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @param Finder $finder
+     * @param DatabaseConnection $db
+     */
+    public function __construct(
+        private Finder $finder,
+        DatabaseConnection $db,
+    ) {
```

Comment:
we should discuss about this but I think you need to repository as they represent places to gather information, I think it's better to have a repository for DBMS, and a Repository for FileSystem, and handle the logic in the useCase